stix2/sources/__init__.py

```diff
@@ -11,104 +11,106 @@
 |
 """
 
+from abc import ABCMeta, abstractmethod
 import uuid
 
+from six import with_metaclass
+
 from stix2.utils import deduplicate
 
 
 def make_id():
     return str(uuid.uuid4())
 
 
-class DataStore(object):
+class DataStore(with_metaclass(ABCMeta)):
     """An implementer will create a concrete subclass from
     this class for the specific DataStore.
 
     Args:
         source (DataSource): An existing DataSource to use
              as this DataStore's DataSource component
-
         sink (DataSink): An existing DataSink to use
              as this DataStore's DataSink component
 
     Attributes:
         id (str): A unique UUIDv4 to identify this DataStore.
-
         source (DataSource): An object that implements DataSource class.
-
         sink (DataSink): An object that implements DataSink class.
 
     """
     def __init__(self, source=None, sink=None):
+        super(DataStore, self).__init__()
         self.id = make_id()
         self.source = source
         self.sink = sink
 
-    def get(self, stix_id, allow_custom=False):
+    @abstractmethod
+    def get(self, stix_id):  # pragma: no cover
```

Comment:
Rather than making these abstract methods, can we just define them similar to `get(self, *args, **kwargs)`, and pass them on the source? I'd rather not require subclasses to override these.